src/components/AnalyticsChart.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+  <div ref="chartContainer"></div>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      chartInstance: {},
+      chartOptions: {
+        title: {
+          text: ''
+        },
+        xAxis: {
+          categories:
+          []
+        },
+        yAxis: {
+          title: {
+            text: ''
+          },
+          plotLines: [{
+            value: 0,
+            width: 1,
+            color: '#808080'
+          }]
+        },
+        navigation: {
+          buttonOptions: {
+            enabled: false
+          }
+        },
+        legend: {
+          layout: 'horizontal',
+          align: 'center',
+          verticalAlign: 'bottom',
+          borderWidth: 0,
+          showInLegend: false
+        },
+        series: [
+          {
+            name: 'Apps Sessions',
+            color: '#03b1fc',
+            data: []
+          },
+          {
+            name: 'Studio sessions',
+            color: '#27799c',
+            data: []
+          }
+        ]
+      }
+    };
+  },
+  props: {
+    appsSessions: {
+      type: Array,
+      default() {
+        return [];
+      }
+    },
+    studioSessions: {
+      type: Array,
+      default() {
+        return [];
+      }
+    }
+  },
+  methods: {
+    initChart: function() {
+      this.transformData()
+        .then(() => {
+          this.chartInstance = Highcharts.chart(this.$refs.chartContainer, this.chartOptions);
+        });
+    },
+    transformData: function() {
+      return new Promise((resolve) => {
```

Comment:
why do you need a promise here?

Reply:
We are using promise here to avoid the init chart before we transfer data to the correct format for Hightcharts to work with.

Reply:
@YaroslavOvdii It looks like everything within `new Promise()` is synchronous though. Are you expecting any of this to be changed to asynchronous when integrated with the API? If not, then this doesn't need to be a promise.

Reply:
@tonytlwu Well no I'm not expecting that. I will remove Promise from that function.